### PR TITLE
feat: redefine enterprise cluster endpoints response behavior

### PR DESCRIPTION
**Pull request by miguelsorianod**
## Description

Related to https://issues.redhat.com/browse/MGDSTRM-10809.

After some discussions with team members around the data returned in the enterprise cluster related endpoints
this PR changes the information they return in the following way:

* The supported instance types and the capacity information is now returned in the Register, Get and
  List Enterprise Cluster endpoints
* Additionally, in the Register enterprise cluster the fleetshard parameters are returned
* The Enterprise Cluster addon parameters endpoint has been changed to now:
  * Return Kind: ClusterAddonParameters
  * Return ID: <cluster_id> of the corresponding Enterprise Clutser
  * Return Href: The URL of the addon parameters endpoint called
  * Return The list of fleetshard parameters
  * Other information about the cluster is not returned anymore

The reason this changes have been made is due to before the set of information returned between the different enterprise cluster endpoints was inconsistent. The reason for that was due to having a set of requirements that have changed over time. This PR makes the behavior between the endpoints consistent and adapted to the current requirements/needs.

## Verification Steps
* Verify enterprise cluster registration endpoint returns the enterprise cluster information, including the fleetshard parameters. Additionally, now it should also include the supported instance types and the capacity information
* Verify both the enterprise cluster get and list endpoints return the enterprise cluster information, now also including the supported instance types and the capacity information. The fleetshard parameters should not be shown as part of the results
* Verify that the enterprise cluster addon parameters endpoint now returns the following
  * kind: "ClusterAddonParameters"
  * id: <cluster_id>
  * href: the url of the addon parameters endpoint called
  * fleetshard_parameters: The list of fleetshard parameters
  * Other information about the cluster itself is not returned anymore

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

**Comment by miguelsorianod** [comment]
Sample response of calling the enterprise cluster get endpoint:
```
{
  "id": "22nht9p37vkq8bknuutb577tkea7i6jl",
  "kind": "Cluster",
  "href": "/api/kafkas_mgmt/v1/clusters/22nht9p37vkq8bknuutb577tkea7i6jl",
  "access_kafkas_via_private_network": false,
  "cluster_id": "22nht9p37vkq8bknuutb577tkea7i6jl",
  "status": "waiting_for_kas_fleetshard_operator",
  "cloud_provider": "aws",
  "region": "us-east-1",
  "multi_az": false,
  "supported_instance_types": {
    "instance_types": [
      {
        "id": "standard",
        "display_name": "Standard",
        "supported_billing_models": [
          {
            "id": "enterprise",
            "ams_resource": "rhosak",
            "ams_product": "RHOSAKCC",
            "ams_billing_models": [
              "standard"
            ]
          }
        ],
        "sizes": [
          {
            "id": "x1",
            "display_name": "1",
            "ingress_throughput_per_sec": {
              "bytes": 52428800
            },
            "egress_throughput_per_sec": {
              "bytes": 104857600
            },
            "total_max_connections": 9000,
            "max_data_retention_size": {
              "bytes": 1073741824000
            },
            "max_partitions": 1500,
            "max_data_retention_period": "P14D",
            "max_connection_attempts_per_sec": 100,
            "max_message_size": {
              "bytes": 1048576
            },
            "min_in_sync_replicas": 2,
            "replication_factor": 3,
            "supported_az_modes": [
              "multi"
            ],
            "quota_consumed": 1,
            "quota_type": "RHOSAK",
            "capacity_consumed": 1,
            "maturity_status": "stable"
          },
          {
            "id": "x2",
            "display_name": "2",
            "ingress_throughput_per_sec": {
              "bytes": 104857600
            },
            "egress_throughput_per_sec": {
              "bytes": 209715200
            },
            "total_max_connections": 18000,
            "max_data_retention_size": {
              "bytes": 2147483648000
            },
            "max_partitions": 3000,
            "max_data_retention_period": "P14D",
            "max_connection_attempts_per_sec": 200,
            "max_message_size": {
              "bytes": 1048576
            },
            "min_in_sync_replicas": 2,
            "replication_factor": 3,
            "supported_az_modes": [
              "multi"
            ],
            "quota_consumed": 2,
            "quota_type": "RHOSAK",
            "capacity_consumed": 2,
            "maturity_status": "preview"
          }
        ]
      }
    ]
  },
  "capacity_information": {
    "kafka_machine_pool_node_count": 3,
    "maximum_kafka_streaming_units": 0,
    "remaining_kafka_streaming_units": 0,
    "consumed_kafka_streaming_units": 0
  }
}
```

Sample response of calling the enterprise cluster list endpoint:
```
{
  "kind": "ClusterList",
  "page": 1,
  "size": 1,
  "total": 1,
  "items": [
    {
      "id": "22nht9p37vkq8bknuutb577tkea7i6jl",
      "kind": "Cluster",
      "href": "/api/kafkas_mgmt/v1/clusters/22nht9p37vkq8bknuutb577tkea7i6jl",
      "access_kafkas_via_private_network": false,
      "cluster_id": "22nht9p37vkq8bknuutb577tkea7i6jl",
      "status": "waiting_for_kas_fleetshard_operator",
      "cloud_provider": "aws",
      "region": "us-east-1",
      "multi_az": false,
      "supported_instance_types": {
        "instance_types": [
          {
            "id": "standard",
            "display_name": "Standard",
            "supported_billing_models": [
              {
                "id": "enterprise",
                "ams_resource": "rhosak",
                "ams_product": "RHOSAKCC",
                "ams_billing_models": [
                  "standard"
                ]
              }
            ],
            "sizes": [
              {
                "id": "x1",
                "display_name": "1",
                "ingress_throughput_per_sec": {
                  "bytes": 52428800
                },
                "egress_throughput_per_sec": {
                  "bytes": 104857600
                },
                "total_max_connections": 9000,
                "max_data_retention_size": {
                  "bytes": 1073741824000
                },
                "max_partitions": 1500,
                "max_data_retention_period": "P14D",
                "max_connection_attempts_per_sec": 100,
                "max_message_size": {
                  "bytes": 1048576
                },
                "min_in_sync_replicas": 2,
                "replication_factor": 3,
                "supported_az_modes": [
                  "multi"
                ],
                "quota_consumed": 1,
                "quota_type": "RHOSAK",
                "capacity_consumed": 1,
                "maturity_status": "stable"
              },
              {
                "id": "x2",
                "display_name": "2",
                "ingress_throughput_per_sec": {
                  "bytes": 104857600
                },
                "egress_throughput_per_sec": {
                  "bytes": 209715200
                },
                "total_max_connections": 18000,
                "max_data_retention_size": {
                  "bytes": 2147483648000
                },
                "max_partitions": 3000,
                "max_data_retention_period": "P14D",
                "max_connection_attempts_per_sec": 200,
                "max_message_size": {
                  "bytes": 1048576
                },
                "min_in_sync_replicas": 2,
                "replication_factor": 3,
                "supported_az_modes": [
                  "multi"
                ],
                "quota_consumed": 2,
                "quota_type": "RHOSAK",
                "capacity_consumed": 2,
                "maturity_status": "preview"
              }
            ]
          }
        ]
      },
      "capacity_information": {
        "kafka_machine_pool_node_count": 3,
        "maximum_kafka_streaming_units": 0,
        "remaining_kafka_streaming_units": 0,
        "consumed_kafka_streaming_units": 0
      }
    }
  ]
}

```

File: internal/kafka/internal/handlers/cluster.go
Comment (machi1990) [inline internal/kafka/internal/handlers/cluster.go:176]: This will result in `1+N` DB calls - we might need to reconsider this in the future when we notice perf issue

**Review by machi1990 — APPROVED** [review]
the change lgtm, I have however left a few comments.

Thanks for adding in integration tests.
I'll let who've have been asked for the review have a look at this as well.

File: openapi/kas-fleet-manager.yaml
Comment (miguelsorianod) [reply openapi/kas-fleet-manager.yaml:2308]: The example was already like this before this PR. I will change it to reflect reality :+1:

File: internal/kafka/internal/handlers/cluster.go
Comment (miguelsorianod) [reply internal/kafka/internal/handlers/cluster.go:176]: :+1: . Could you open a jira issue to keep track of that?

Thanks

Comment (miguelsorianod) [reply internal/kafka/internal/handlers/cluster.go:176]: Thanks :+1:

File: openapi/kas-fleet-manager.yaml
Comment (machi1990) [inline openapi/kas-fleet-manager.yaml:2334]:
```suggestion
```

Comment (machi1990) [reply openapi/kas-fleet-manager.yaml:2334]: You could further remove this if we want to be precise as standard instance can only be deployed on multi_az clusters and are thus multi_az.

File: openapi/kas-fleet-manager.yaml
Comment (machi1990) [inline openapi/kas-fleet-manager.yaml:2385]:
```suggestion
```

File: openapi/kas-fleet-manager.yaml
Comment (miguelsorianod) [reply openapi/kas-fleet-manager.yaml:2334]: I will change it :+1:

Comment (miguelsorianod) [reply openapi/kas-fleet-manager.yaml]: Updated

**Review by machi1990 — APPROVED** [review]
LGTM

thanks @miguelsorianod

**Review by dimakis — APPROVED** [review]
LGTM, 
I won't get the chance to verify it though

File: openapi/kas-fleet-manager.yaml
Comment (machi1990) [reply openapi/kas-fleet-manager.yaml]: thanks

**Comment by machi1990** [comment]
@miguelsorianod is this waiting for another round of reviews?

**Comment by miguelsorianod** [comment]
No. This can be merged.

**Pull request merged**
